lib/bfd.c

```diff
@@ -1334,3 +1334,8 @@ int bfd_nht_update(const struct prefix *match, const struct zapi_route *route)
 
 	return 0;
 }
+
+bool bfd_session_is_shutdown(const struct bfd_session_params *session)
+{
+	return session->bss.state == BSS_ADMIN_DOWN;
```

Comment:
Should BSS_DOWN be also added to this function?

IMO, both BSS_ADMIN_DOWN  and BSS_DOWN must be covered.
If my BFD session is down, BGP should remain down too

Reply:
I think yes, will change slightly current approach.

Reply:
then let's do it right. I can test once this is also resolve. Thank you very much for this prompt response.

Reply:
I changed the logic to the if the session is really UP.